cli/listener/src/main.rs

```diff
@@ -1,32 +1,41 @@
-use chrono::TimeZone;
-use clap::Clap;
-use lido::{
-    state::Lido,
-    token::{ArithmeticError, Rational},
+use std::{
+    fmt,
+    sync::{Arc, Mutex},
+    thread::JoinHandle,
+    time::{Duration, Instant},
 };
+
+use clap::Clap;
+use lido::token::Rational;
+use rand::{rngs::ThreadRng, Rng};
 use rusqlite::{params, Connection, Row};
+use serde::Serialize;
 use solana_client::rpc_client::RpcClient;
 use solana_sdk::{
     clock::{Epoch, Slot},
     commitment_config::CommitmentConfig,
     pubkey::Pubkey,
     signature::Keypair,
 };
-use solido_cli_common::snapshot::{
-    Config, OutputMode, SnapshotClient, SnapshotConfig, SnapshotError,
+use solido_cli_common::{
+    error::Error,
+    snapshot::{Config, OutputMode, SnapshotClient, SnapshotClientConfig},
 };
+use tiny_http::{Header, Request, Response, Server};
 
 #[derive(Clap, Debug)]
 pub struct Opts {
+    /// Solido's instance address
+    #[clap(long)]
+    solido_address: Pubkey,
+
     /// URL of cluster to connect to (e.g., https://api.devnet.solana.com for solana devnet)
-    // Overwritten by `GeneralOpts` if None.
     #[clap(long, default_value = "http://127.0.0.1:8899")]
     cluster: String,
 
     /// Whether to output text or json. [default: "text"]
-    // Overwritten by `GeneralOpts` if None.
     #[clap(long = "output", possible_values = &["text", "json"])]
-    output_mode: Option<OutputMode>,
+    output_mode: OutputMode,
```

Comment:
Hmm now that I’m looking at this again ... why do we have this here? For `solido` one-off commands it makes sense, but for just a daemon that watches the chain ... it shouldn’t output anything aside from logging and for that we don’t need this switch right now. Is the `output_mode` used at all in this crate?

Reply:
I think would be nice to show some log, but that can be done just with some `println` 👍

Reply:
Yeah exactly, and we can even still have the enum for output instead of raw printlns in the middle of the code if you like, but no need to have it be switchable between `Display` and `serde::Serialize`, just `Display` is sufficient.